OpenLaTeXStudio/EditorModule/src/latexstudio/editor/EditorTopComponent.java

```diff
@@ -1,308 +1,322 @@
-        urls[2] = getClass().getResource("/latexstudio/editor/resources/latex-mathsymbols.cwl");
-
-        for (URL url : urls) {
-            InputStream is = null;
-            try {
-                is = url.openStream();
-                BufferedReader br = new BufferedReader(new InputStreamReader(is));
-                String line;
-                while ((line = br.readLine()) != null) {
-                    if (!line.startsWith("#")) {
-                        provider.addCompletion(new BasicCompletion(provider, line.substring(1)));
-                    }
-                }
-            } catch (IOException ex) {
-                Exceptions.printStackTrace(ex);
-            } finally {
-                IOUtils.closeQuietly(is);
-            }
-        }
-
-        return provider;
-    }
-}
+/* 
+ * Copyright (c) 2015 Sebastian Brudzinski
+ * 
+ * See the file LICENSE for copying permission.
+ */
+package latexstudio.editor;
+
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+import javax.swing.text.BadLocationException;
+import latexstudio.editor.remote.DbxState;
+import latexstudio.editor.settings.ApplicationSettings;
+import latexstudio.editor.util.ApplicationUtils;
+import org.apache.commons.io.IOUtils;
+import org.fife.ui.autocomplete.AutoCompletion;
+import org.fife.ui.autocomplete.BasicCompletion;
+import org.fife.ui.autocomplete.CompletionProvider;
+import org.fife.ui.autocomplete.DefaultCompletionProvider;
+import org.netbeans.api.settings.ConvertAsProperties;
+import org.openide.awt.ActionID;
+import org.openide.awt.ActionReference;
+import org.openide.util.Exceptions;
+import org.openide.util.NbBundle.Messages;
+import org.openide.windows.TopComponent;
+
+/**
+ * Top component which displays the editor window.
+ */
+@ConvertAsProperties(
+        dtd = "-//latexstudio.editor//Editor//EN",
+        autostore = false
+)
+@TopComponent.Description(
+        preferredID = "EditorTopComponent",
+        //iconBase="SET/PATH/TO/ICON/HERE", 
+        persistenceType = TopComponent.PERSISTENCE_ALWAYS
+)
+@TopComponent.Registration(mode = "editor", openAtStartup = true)
+@ActionID(category = "Window", id = "latexstudio.editor.EditorTopComponent")
+@ActionReference(path = "Menu/Window" /*, position = 333 */)
+@TopComponent.OpenActionRegistration(
+        displayName = "#CTL_EditorAction",
+        preferredID = "EditorTopComponent"
+)
+@Messages({
+    "CTL_EditorAction=Editor",
+    "CTL_EditorTopComponent=Editor Window",
+    "HINT_EditorTopComponent=This is a Editor window"
+})
+public final class EditorTopComponent extends TopComponent {
+
+    private boolean dirty = false;
+    private File currentFile;
+    private DbxState dbxState;
+
+    private static final String AUTO_COMPLETE_DELAY_PROPERTY = "auto.complete.delay";
```

Comment:
I suppose for now, we may make ApplicationSettings constants public, which is not the best, but still better than having this duplicate. Please use the property name constant from there, unless You, @sabbir044 have got some better idea on how to improve this.